include/exec/__detail/__bit_cast.hpp

```diff
@@ -16,6 +16,8 @@
  */
 #pragma once
 
+#include <cstring>
```

Comment:
Not related to this PR, but I noticed that the fallback implementation of `bit_cast()` is marked constexpr (which matches the std implementation) but is using std::memcpy, which isn't marked constexpr and so won't compile if actually invoked in a constexpr context.

Maybe it should be implemented in terms of the respective compiler intrinsics instead?
e.g. `__builtin_bit_cast()` on clang
Or not marked constexpr?

Reply:
I've left the `constexpr` but changed the implementation to use `__builtin_bit_cast` when available.